tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/GivenSyntax36Suite.scala

```diff
@@ -314,4 +314,553 @@ class GivenSyntax36Suite extends BaseDottySuite {
+    val code = """|given ord: ([A] => (Ord[A]) => Ord[List[A]]) => Ord[List[A]]:
+                  |  def foo = ???
+                  |""".stripMargin
+    val layout = "given ord: ([A] => Ord[A] => Ord[List[A]]) => Ord[List[A]] { def foo = ??? }"
+    val tree = Defn.Given(
+      Nil,
+      "ord",
+      Nil,
+      List(List(tparam(
+        "",
+        ppolyfunc(pfunc(papply("Ord", papply("List", "A")), papply("Ord", "A")), pparam(Nil, "A"))
+      ))),
+      tpl(List(init(papply("Ord", papply("List", "A")))), List(Defn.Def(Nil, "foo", Nil, None, "???")))
+    )
+    runTestAssert[Stat](code, layout)(tree)
+  }
+
+  test("context a poly function, braces") {
+    val code = """|given ord: ([A] => (Ord[A]) => Ord[List[A]]) => Ord[List[A]] {
+                  |  def foo = ???
+                  |}
+                  |""".stripMargin
+    val layout = "given ord: ([A] => Ord[A] => Ord[List[A]]) => Ord[List[A]] { def foo = ??? }"
+    val tree = Defn.Given(
+      Nil,
+      "ord",
+      Nil,
+      List(List(tparam(
+        "",
+        ppolyfunc(pfunc(papply("Ord", papply("List", "A")), papply("Ord", "A")), pparam(Nil, "A"))
+      ))),
+      tpl(List(init(papply("Ord", papply("List", "A")))), List(Defn.Def(Nil, "foo", Nil, None, "???")))
+    )
+    runTestAssert[Stat](code, layout)(tree)
+  }
+
+  test("given type a function, coloneol") {
+    val code = """|given ord: [A] => (Ord[A] => Ord[List[A]]):
+                  |  def foo = ???
+                  |""".stripMargin
+    val layout = "given ord[A]: (Ord[A] => Ord[List[A]]) with { def foo = ??? }"
+    val tree = Defn.Given(
+      Nil,
+      "ord",
+      List(pparam(Nil, "A")),
+      Nil,
+      tpl(
+        List(init(pfunc(papply("Ord", papply("List", "A")), papply("Ord", "A")))),
+        List(Defn.Def(Nil, "foo", Nil, None, "???"))
+      )
+    )
+    runTestAssert[Stat](code, layout)(tree)
+  }
+
+  test("given type a function, braces") {
+    val code = """|given ord: [A] => (Ord[A] => Ord[List[A]]) {
+                  |  def foo = ???
+                  |}
+                  |""".stripMargin
+    val layout = "given ord[A]: (Ord[A] => Ord[List[A]]) with { def foo = ??? }"
+    val tree = Defn.Given(
+      Nil,
+      "ord",
+      List(pparam(Nil, "A")),
+      Nil,
+      tpl(
+        List(init(pfunc(papply("Ord", papply("List", "A")), papply("Ord", "A")))),
+        List(Defn.Def(Nil, "foo", Nil, None, "???"))
+      )
+    )
+    runTestAssert[Stat](code, layout)(tree)
+  }
+
+  // extra new syntax tests: param clause first
+
+  test("named, param clause first, coloneol") {
+    val code = """|given ord: (a: A) => Ord[A] => Ord[List[A]]:
+                  |  def foo = ???
+                  |""".stripMargin
+    val layout = "given ord: (a: A) => (Ord[A]) => Ord[List[A]] { def foo = ??? }"
+    val tree = Defn.Given(
+      Nil,
+      "ord",
+      Nil,
+      List(List(tparam("a", "A")), List(tparam("", papply("Ord", "A")))),
```

Comment:
Shouldn't we show this as a type lambda instead?

Reply:
a type lambda, i think, uses the `=>>` arrow. 

but in `given ord: (a: A) => Ord[A] => Ord[List[A]]`, you have a name (`ord`), two conditions (`(a: A)` and `Ord[A]`, i think they are interpreted as having their own `given` defined) and the type of the `given` expression (`Ord[List[A]]`).

none of the types above are lambdas or functions.

is your question why i didn't encode these conditions using a lambda, instead of type or term param clauses?

Reply:
Yeah, I am still thinking about whether that would even make sense to encode those as lambdas

Reply:
the existing trees used param clauses, so using anything else would require creating completely new trees; we can safely expand from one param clause to many but that's it. 

but the published ebnf uses, specifically, these four as GivenConditional:
- TypeParamClause (for [])
- TermParamClause (for () with param names)
- FunArgTypes (for () without param names)
- or an infix type but without refinements

so I've simply selected to use the first two and allow params not to have names (i e., anonymous names), which we already supported with `using` clauses anyway.

i thought it was the simplest, least dramatic change.